Sources/DVR/Filter.swift

```diff
@@ -140,13 +140,18 @@ open class Filter {
         // TODO: needs to account for different ways of encoding form data
     }
 
-    final func filter(request: URLRequest) -> URLRequest {
-        var filtered = request
+    final func filter(request: URLRequest?) -> URLRequest? {
+        
+        guard var filtered = request else {
+            return nil
+        }
         filterHeaders(for: &filtered)
         filterQueryParams(for: &filtered)
         filterPostParams(for: &filtered)
-        filtered = beforeRecordRequest?(filtered) ?? filtered
-        return filtered
+        guard beforeRecordRequest != nil else {
+            return filtered
+        }
+        return beforeRecordRequest!(filtered)
```

Comment:
Forced unwrapping should be avoided.

```suggestion
        if let beforeRecordRequest = beforeRecordRequest  {
            return beforeRecordRequest(filtered)
        }
        return filtered
```